tools/test/test-setup.sh

```diff
@@ -280,15 +302,16 @@ cleanupPid=$!
 
 set +m
 
-# Wait until $childPid fully exits.
+# Wait until child process fully exits.
 # We need to wait in a loop because wait is interrupted by any incoming trapped
 # signal (https://www.gnu.org/software/bash/manual/bash.html#Signals).
 while kill -0 $childPid 2>/dev/null; do
+  if [[ "$targetExited" == "1" ]]; then
+    . "${targetExitCodeFile}"
```

Comment:
I think this deserves a comment that this will read the exit code from the file.

Btw, is there a scenario where we didn't write the file?